### PR TITLE
Fix render mesh issue in native engine.

### PR DESCRIPTION
**Pull request by hana-alice**
in engine/........./mesh.ts:
```
  if (dstStride === 4 && !gfxDevice.hasFeature(Feature.ELEMENT_INDEX_UINT)) {
      const vertexCount = this._struct.vertexBundles[prim.vertexBundelIndices[0]].view.count;
      if (vertexCount >= 65536) {
          warnID(10001, vertexCount, 65536);
          console.log('prim.ignored');
          continue; // Ignore this primitive
      } else {
          dstStride >>= 1; // Reduce to short.
          dstSize >>= 1;
      }
  }
```

this one `gfxDevice.hasFeature(Feature.ELEMENT_INDEX_UINT)` matters

### CODE REVIEW

**Comment by hana-alice** [comment]
@minggo @pandamicro

File: cocos/renderer/gfx-metal/MTLCommandBuffer.mm [hunk 1]
```diff
@@ -215,6 +215,7 @@ of this software and associated documentation files (the "Software"), to deal
             if (_indirectDrawSuppotred) {
                 ++_numDrawCalls;
                 if (indirectBuffer->isDrawIndirectByIndex()) {
+                    _indexType = indexBuffer->getIndexType();
```
Comment (minggo) [inline cocos/renderer/gfx-metal/MTLCommandBuffer.mm:218]: If fetch index type in draw, then should record it in command buffer?

File: cocos/renderer/gfx-gles2/GLES2Device.cpp [hunk 2]
```diff
@@ -95,6 +95,7 @@ bool GLES2Device::initialize(const DeviceInfo &info) {
     _features[(int)Feature::FORMAT_R11G11B10F] = true;
     _features[(int)Feature::FORMAT_D24S8] = true;
     _features[(int)Feature::MSAA] = true;
+    _features[(uint)Feature::ELEMENT_INDEX_UINT] = GL_OES_element_index_uint;
```
Comment (YunHsiao) [inline cocos/renderer/gfx-gles2/GLES2Device.cpp:98]: `GL_OES_element_index_uint` is a macro that is always defined to be true. use runtime checking instead:
```
if (checkExtension("GL_OES_element_index_uint")) {
    _features[(int)Feature::ELEMENT_INDEX_UINT] = true;
}
```

File: cocos/renderer/gfx-gles3/GLES3Device.cpp [hunk 3]
```diff
@@ -92,6 +92,7 @@ bool GLES3Device::initialize(const DeviceInfo &info) {
     _features[(int)Feature::INSTANCED_ARRAYS] = true;
     _features[(int)Feature::MULTIPLE_RENDER_TARGETS] = true;
     _features[(uint)Feature::BLEND_MINMAX] = true;
+    _features[(uint)Feature::ELEMENT_INDEX_UINT] = GL_OES_element_index_uint;
```
Comment (YunHsiao) [inline cocos/renderer/gfx-gles3/GLES3Device.cpp:95]: ES3 标准直接支持，直接赋值就好

**Review by minggo — APPROVED** [review]

**Review by YunHsiao — APPROVED** [review]

**Pull request merged**